benchmark/atoi-corpus.h

```diff
@@ -0,0 +1,70 @@
+#include <vector>
+#include <string>
+#include <random>
+
+struct Corpus8DecimalDigits {
+    std::vector<int> asNumbers_;
+    std::string characters_;
+
+    Corpus8DecimalDigits(std::vector<int> aNs, std::string cs):
+        asNumbers_(aNs),
+        characters_(cs)
+    {}
+
+    template<typename G>
+    static auto makeCorpus(G &generator) {
+        auto count = 1031; // 1031 is a prime number, this helps to disable in
+        // practice the branch predictor, the idea is to measure the performance
+        // of the code under measurement, not how the the unrealistic conditions
+        // of microbenchmarking help/hurt the code under measurement
+        std::string allCharacters;
+        allCharacters.resize(count * 9);
+        std::vector<int> inputs;
+        std::uniform_int_distribution<> range(0, 100*1000*1000 - 1);
+        char *base = allCharacters.data();
+        for(;;) {
+            auto input = range(generator);
+            snprintf(base, 9, "%08d", input);
+            inputs.push_back(input);
+            if(--count) { break; }
+            base += 9;
+        }
+        return Corpus8DecimalDigits(inputs, allCharacters);
+    }
+
+    struct Iterator {
+        Corpus8DecimalDigits *thy;
+        int *ip;
+        char *cp;
```

Comment:
I don't understand the names of these variables. Are there more friendly names for these possible?

Reply:
Integer pointer and character pointer.
I have the policy of choosing deliberately short identifiers, especially if they feel cryptic, to indicate "this name does not matter".  These names do not matter, since their role is immediately apparent by each use.  Please accept this element of style and "resolve" this comment to indicate it.

Reply:
I tend to advocate for verbose names, even if that is something like `tempVal` (I'm sure you could think of something better than `tempVal`), otherwise readers might think they are missing some context. However, if this is already and established convention in the code, then it should probably continue. Just my two cents!

Reply:
"otherwise readers might think":
Unfortunately, coding with an audience in mind that may include very superficial readers is limitative, it is a drag on the authors and contributors.
This codebase requires readers to understand the concept of negative space, things that may even be conspicuous because of their absence.
There used to be a great presentation about Post Modern C++ by Tony Van Eerd at Avast in Prague, that provides a great introduction to the concept of negative space in creative works, but it seems it is not available anymore.  Please watch this presentation, that it is in the same subject and still very good, but not necessarily directly applicable to this thread (I can't remember):
https://youtu.be/QTLn3goa3A8

Reply:
I generally like the 'this name is semantically void' convention E uses in this repo.

Reply:
Though I admit it was jarring through unfamiliarity at first.